zqd/packet/packet.go

```diff
@@ -74,64 +74,83 @@ func IngestFile(ctx context.Context, s *space.Space, pcap, zeekExec string) (*In
 		done:      make(chan struct{}),
 		zeekExec:  zeekExec,
 	}
+	if err = p.indexPcap(ctx); err != nil {
+		os.Remove(p.space.DataPath(IndexFile))
+		return nil, err
+	}
+	if err = p.space.SetPacketPath(p.pcapPath); err != nil {
+		os.Remove(p.space.DataPath(IndexFile))
+		return nil, err
+	}
 	go func() {
 		p.err = p.run(ctx)
 		close(p.done)
 	}()
-	return p, p.awaitZeekLogs()
+	return p, nil
 }
 
 func (p *IngestProcess) run(ctx context.Context) error {
-	idx, err := p.slurp(ctx)
-	if err != nil {
-		goto abort
+	var slurpErr error
+	slurpDone := make(chan struct{})
+	go func() {
+		slurpErr = p.slurp(ctx)
+		close(slurpDone)
+	}()
+
+	abort := func() {
+		os.RemoveAll(p.logdir)
+		os.Remove(p.space.DataPath(IndexFile))
+		p.space.SetPacketPath("")
 	}
-	if err = p.writeIndexFile(idx); err != nil {
-		goto abort
+
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+	start := time.Now()
+	next := time.Second
+outer:
+	for {
+		select {
+		case <-slurpDone:
+			break outer
+		case t := <-ticker.C:
+			if t.After(start.Add(next)) {
+				if err := p.writeData(ctx); err != nil {
```

Comment:
writeData() is a vague name.  How about createSnapshot()?

Reply:
I agree and there's a bunch of other names around here that I think deserve adjusting, but I was going to keep this for a separate PR... I'll adjust writeData in this one.

Reply:
ok, leave it

Reply:
changed